wgpu-core/src/command/bundle.rs

```diff
@@ -1036,14 +1040,20 @@ pub mod bundle_ffi {
         stages: wgt::ShaderStage,
         offset: u32,
         size_bytes: u32,
-        data: *const u32,
+        data: *const u8,
     ) {
         span!(_guard, DEBUG, "RenderBundle::set_push_constants");
-        let data_slice = slice::from_raw_parts(data, (size_bytes / 4) as usize);
+        let data_slice = slice::from_raw_parts(data, size_bytes as usize);
         let value_offset = pass.base.push_constant_data.len().try_into().expect(
             "Ran out of push constant space. Don't set 4gb of push constants per RenderBundle.",
         );
+
+        let aligned_len = ((data_slice.len() + 3) / 4) * 4;
+        pass.base.push_constant_data.reserve(aligned_len);
```

Comment:
instead of doing this, we could just do `pass.base.push_constant_data.resize(aligned_len, 0);`

Reply:
Well I thought I don't have to zero the memory that way.